public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilder.tsx

```diff
@@ -83,43 +84,89 @@ export const PromQueryBuilder = React.memo<Props>((props) => {
       .map((k) => ({ value: k }));
   };
 
+  const getLabelValuesAutocompleteSuggestions = (
+    queryString?: string,
+    labelName?: string
+  ): Promise<SelectableValue[]> => {
+    const forLabel = {
+      label: labelName ?? '__name__',
+      op: '=~',
+      value: regexifyLabelValuesQueryString(`${queryString}`),
```

Comment:
Ok, so I've been worried about calling this function for this field (this function removes any whitespace in the queries text and adds a wildcard selector) allows users to enter in their own regex. The argument for keeping it is that whitespace is never valid in a label name or value, so whenever the user enters a whitespace char they won't get back a valid query. But is there any regex query that could be executed that contains whitespace that we don't want broken up? Whitespace is just used to match whitespace, so again my argument is that this should be safe, but I want more confidence.